### PR TITLE
feat: add AWS specific requirements

### LINKED ISSUE #6990: Another mechanism for IAM Roles for Service Account creation in EKS
### Summary 

Another mechanism for IAM Roles for Service Account creation in EKS

### Steps to reproduce the behavior

1. Initiate `jxl` installation using `jxl boot create`
1. Install Jenkins X using `jxl boot run --git-url <dev-repo-url>` (created in step 1)

### Expected behavior

Jenkins X installation creates all necessary application helm charts

### Actual behavior

Because the `jxl boot create` step creates IAM Roles and Service Accounts, helm charts that try to create those SAs fail. There's a `--terraform` arg to avoid creating IRSA and other dependent cloud resources, but if SAs are created using whatever tooling before helm charts are installed there will be resource conflicts.

It is possible to use templating to skip creation of SAs using helm, but it might just be better to use helm to manage k8s resources. Additional properties in some SAs such as this one: https://github.com/jenkins-x-charts/tekton/blob/master/tekton/templates/251-bot-serviceaccount.yaml#L22-L27 will have to also be accounted for in cloud tooling to create IRSA

### Jx version

The output of `jx version` is:

`jxl version`
NAME               VERSION
jxl                0.0.150
Kubernetes cluster v1.14.9-eks-502bfb
kubectl            v1.17.3
helm client        3.1.2
git                2.24.1
Operating System   "Fedora release 31 (Thirty One)"

### Jenkins type

- [x] Serverless Jenkins X Pipelines (Tekton + Lighthouse)
- [ ] Classic Jenkins

### Kubernetes cluster

EKS 1.14 using eksctl

### Operating system / Environment

Operating System   "Fedora release 31 (Thirty One)"

### PR DESCRIPTION
**Pull request by gazal-k**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [x] Change is covered by existing or new tests.

#### Which issue this PR fixes

fixes #6990

### CODE REVIEW

**Comment by hferentschik** [comment]
@gazal-k, thanks for the pull request. Can you provide some context what issue or usecase you are trying to solve? Maybe you could create an [issue](https://github.com/jenkins-x/jx/issues) for it.

I am trying to determine what you are trying to achieve by modifying the boot requirements struct w/o any further changes to make code to make use of it.

**Comment by gazal-k** [comment]
@hferentschik I'm not entirely sure what I'm doing. I mean, I'm not sure how to fix these broken builds. What I'm trying to do essentially is, fix this issue: https://github.com/jenkins-x-labs/issues/issues/17

File: pkg/config/install_requirements.go [hunk 1]
```diff
@@ -316,6 +316,14 @@ type StorageConfig struct {
 	Backup StorageEntryConfig `json:"backup"`
 }
 
+// AWSConfig contains AWS specific requirements
+type AWSConfig struct {
+	// AccountID the unique 12-digit number assigned to your AWS account
+	AccountID string `json:"accountId,omitempty"`
+	// UseIRSAAnnotations if true, IAM roles will be assumed to exist and they'll be linked to ServiceAccounts using annotations
+	UseIRSAAnnotations bool `json:"useIRSAAnnotations"`
```
Comment (hferentschik) [inline pkg/config/install_requirements.go]: You could try to add `omitempty` here as well.

**Comment by hferentschik** [comment]
Here is the unit test failure:

```
      Expected
          <string>: gcr.io/jenkinsxio-labs/**l:0.0.127
      to equal
          <string>: gcr.io/jenkinsxio/builder-maven:0.1.527
      /workspace/source/pkg/tekton/metapipeline/metapipeline_test.go:191
------------------------------
••••••
Summarizing 2 Failures:
[Fail] Meta pipeline #createMetaPipelineCRDs with extending App [It] contain three task steps 
/workspace/source/pkg/tekton/metapipeline/metapipeline_test.go:185
[Fail] Meta pipeline #createMetaPipelineCRDs with extending App [It] uses version stream for default image 
/workspace/source/pkg/tekton/metapipeline/metapipeline_test.go:191
Ran 23 of 23 Specs in 0.152 seconds
FAIL! -- 21 Passed | 2 Failed | 0 Pending | 0 Skipped
--- FAIL: TestMetaPipeline (0.15s)
FAIL
FAIL	github.com/jenkins-x/**/pkg/tekton/metapipeline	7.572s
ok  	github.com/jenkins-x/**/pkg/tekton/syntax	0.597s
?   	github.com/jenkins-x/**/pkg/tekton/syntax/syntax_helpers_test	[no test files]
?   	github.com/jenkins-x/**/pkg/tekton/tekton_helpers_test	[no test files]
ok  	github.com/jenkins-x/**/pkg/terraform	0.008s
?   	github.com/jenkins-x/**/pkg/testkube	[no test files]
ok  	github.com/jenkins-x/**/pkg/tests	0.021s
ok  	github.com/jenkins-x/**/pkg/users	0.112s
ok  	github.com/jenkins-x/**/pkg/util	10.395s
ok  	github.com/jenkins-x/**/pkg/util/json	0.014s
?   	github.com/jenkins-x/**/pkg/util/mocks	[no test files]
?   	github.com/jenkins-x/**/pkg/util/secrets	[no test files]
ok  	github.com/jenkins-x/**/pkg/util/system	0.006s
?   	github.com/jenkins-x/**/pkg/util/trace	[no test files]
ok  	github.com/jenkins-x/**/pkg/vault	0.014s
?   	github.com/jenkins-x/**/pkg/vault/create	[no test files]
?   	github.com/jenkins-x/**/pkg/vault/fake	[no test files]
?   	github.com/jenkins-x/**/pkg/vault/mocks	[no test files]
?   	github.com/jenkins-x/**/pkg/vault/mocks/matchers	[no test files]
ok  	github.com/jenkins-x/**/pkg/version	0.005s
ok  	github.com/jenkins-x/**/pkg/versionstream	0.018s
?   	github.com/jenkins-x/**/pkg/versionstream/versionstreamrepo	[no test files]
?   	github.com/jenkins-x/**/pkg/virtualmachines/hyperkit	[no test files]
?   	github.com/jenkins-x/**/pkg/virtualmachines/kvm	[no test files]
?   	github.com/jenkins-x/**/pkg/virtualmachines/kvm2	[no test files]
?   	github.com/jenkins-x/**/pkg/virtualmachines/virtualbox	[no test files]
?   	github.com/jenkins-x/**/pkg/workflow	[no test files]
make: *** [Makefile:105: test] Error 1
```

File: pkg/config/install_requirements.go (hunk 1 above)
Comment (gazal-k) [reply pkg/config/install_requirements.go]: I hadn't noticed that for the other `bool` fields. I thought perhaps it was not needed

**Comment by gazal-k** [comment]
I got the same error running `make lint` on master as on my feature branch:

```sh
FAILED make verify-generation-complete
RUN make generate-docs
make[1]: Entering directory '/home/mgazal/go/src/github.com/jenkins-x/jx'
Generating HTML docs for Kubernetes Clients
./build/codegen --generator-version 9642bd3f4de5 docs
INFO[0000] generating docs to /home/mgazal/go/src/github.com/jenkins-x/jx/docs/apidocs  app=codegen
INFO[0000] installing github.com/kubernetes-incubator/reference-docs/gen-apidocs in version 9642bd3f4de5 via 'go get'  app=codegen
INFO[0002] copying static files from /home/mgazal/.jx/codegen/go/pkg/mod/github.com/kubernetes-incubator/reference-docs@v0.0.0-20190410022412-9642bd3f4de5/gen-apidocs/generators/static to /home/mgazal/go/src/github.com/jenkins-x/jx/docs/apidocs/static  app=codegen
make[1]: Leaving directory '/home/mgazal/go/src/github.com/jenkins-x/jx'
PASSED make generate-docs
make: *** [Makefile:216: lint] Error 1
```
I suppose lint is just broken and not as a result of this change

**Comment by gazal-k** [comment]
> I got the same error running `make lint` on master as on my feature branch:
> 
> ```shell
> FAILED make verify-generation-complete
> RUN make generate-docs
> make[1]: Entering directory '/home/mgazal/go/src/github.com/jenkins-x/jx'
> Generating HTML docs for Kubernetes Clients
> ./build/codegen --generator-version 9642bd3f4de5 docs
> INFO[0000] generating docs to /home/mgazal/go/src/github.com/jenkins-x/jx/docs/apidocs  app=codegen
> INFO[0000] installing github.com/kubernetes-incubator/reference-docs/gen-apidocs in version 9642bd3f4de5 via 'go get'  app=codegen
> INFO[0002] copying static files from /home/mgazal/.jx/codegen/go/pkg/mod/github.com/kubernetes-incubator/reference-docs@v0.0.0-20190410022412-9642bd3f4de5/gen-apidocs/generators/static to /home/mgazal/go/src/github.com/jenkins-x/jx/docs/apidocs/static  app=codegen
> make[1]: Leaving directory '/home/mgazal/go/src/github.com/jenkins-x/jx'
> PASSED make generate-docs
> make: *** [Makefile:216: lint] Error 1
> ```
> 
> I suppose lint is just broken and not as a result of this change

that was just some GOPATH issues. Figured out why lint was broken

**Comment by hferentschik** [comment]
@gazal-k, I am going to close this pull request. This change is really targeting jxl and the multicluster branch. These were really experiments. There is ongoing work for implementing multicluster right now and for other parts of the experiment we need to see how we move forward. IMO it would feel premature to merge this atm.

**Comment by gazal-k** [comment]
no worries @hferentschik . I created another issue: https://github.com/jenkins-x/terraform-aws-eks-jx/issues/31 so we can continue the discussion and perhaps come up with a better solution

**Pull request closed without merging**